lib/collection/src/collection_manager/optimizers/mod.rs

```diff
@@ -1,5 +1,177 @@
+use std::collections::VecDeque;
+use std::sync::Arc;
+
+use chrono::{DateTime, Utc};
+use parking_lot::Mutex;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use super::holders::segment_holder::SegmentId;
+
 pub mod config_mismatch_optimizer;
 pub mod indexing_optimizer;
 pub mod merge_optimizer;
 pub mod segment_optimizer;
 pub mod vacuum_optimizer;
+
+/// Number of last trackers to keep in tracker log
+///
+/// Will never remove older trackers for failed or still ongoing optimizations.
```

Comment:
Is it possible to have a very large number of failed optimizations information here as we keep them forever?

Reply:
Yes, errors are only removed on restart.

Note that if an optimization panics it is highly recommended to restart the server anyway. The collection status will be red until a restart as well. So I'm not afraid of this growing indefinitely.